retrieval/scrape.go

```diff
@@ -833,12 +877,9 @@ func (sl *scrapeLoop) reportStale(start time.Time) error {
 }
 
 func (sl *scrapeLoop) addReportSample(app storage.Appender, s string, t int64, v float64) error {
-	re, ok := sl.refCache[s]
-
+	ref, ok := sl.cache.getRef(s)
```

Comment:
I don't think it's safe to share the cache here, as we're using a different appender and thus the same input series might result in a different storage series.